### ISSUE TITLE
Error with using $url addMedia

### ISSUE THREAD

**Issue by adamroyle** [jeroendesloovere/vcard#87]
This change https://github.com/jeroendesloovere/vcard/commit/ea119a3de27ddb12cd7b71d5cd769e040dfc7909 has caused issues on php 5.6 in my app. I get the following error:

```
Warning (2): mime_content_type(): Failed identify data 0:no magic files loaded in ...
```

I'm guessing because I'm passing a url instead of a file location, and mime_content_type() doesn't automatically download the file?

**Comment by claytoncollie** [comment]
for a quick fix change line 218 to $mimetype = 'image/jpeg'; if you are adding a jpg of course.

**Comment by DevNIX** [comment]
Why not trying something like this? https://github.com/trippo/ResponsiveFilemanager/blob/master/filemanager/include/mime_type_lib.php

**Comment by adamroyle** [comment]
The problem is not in detecting the mime type - the problem is the regression in functionality. The previous code would detect the mime type from a base64'd value of the data, whereas the new code uses `mime_content_type()` of the URL, which doesn't work because it's a url and not a file location.

I'm not sure why this code was changed, but I see the following options:

1. Revert this change **OR**
2. Save the file contents to a temporary file before using mime_content_type()

**Comment by ApeWare** [comment]
Doing some updates and ran into this exact issue. We store our images on s3 and pass the url to `addPhoto($url)` and now we are getting the mime_content_type() error. 

Is there a reason for this breaking change other than to reduce the library by 3 lines of code?  Debating on forking and rolling back this change, but would prefer to understand why the change was important.

**Comment by ifunk** [comment]
Instead of forking we have just fixed our version at 1.2.2. At this time there's no need for us to upgrade to a newer version as we've been running it in production for over a year and have had no issues.

**Comment by axelitus** [comment]
I have the same issue. `mime_content_type()` does not work with urls, just local filenames. If the media contents has been already downloaded why not use that to verify the mime_type and do another request just to get the mime_type? I don't understand this change...
